scripts/creation-portal-migration/template/generate/gQuestion.js

```diff
@@ -0,0 +1,511 @@
+const { questionTemplate, questionStatic } = require("../config/question");
+
+/**
+ * To map the body field data
+ * @method
+ * @name getQuestionBodyParagraph
+ * @param {Object} questionData - questionData
+ * @returns {String} - returns mapped body field data with p tag
+ **/
+const getQuestionBodyParagraph = (questionData) => {
+  return questionData
+    .map((data) => {
+      return `<p>${data}</p>`;
+    })
+    .join("");
+};
+
+/**
+ * To map the body field data  with div tag
+ * @method
+ * @name getQuestionBodyDiv
+ * @param {Object} questionData - questionData
+ * @returns {String} - returns mapped body field data with div tag
+ **/
+const getQuestionBodyDiv = (questionData) => {
+  const divs = questionData
+    .map((data) => {
+      return `<div class='mcq-title'><p>${data}&nbsp</p></div><div data-choice-interaction='response1' class='mcq-vertical'></div>`;
+    })
+    .join("");
+  const div = `<div class='question-body'>${divs}</div>`;
+  return div;
+};
+
+/**
+ * To map the question type date
+ * @method
+ * @name getDateTemplate
+ * @param {Object} question - question
+ * @returns {Object} - returns mapped date type question
+ **/
+const getDateTemplate = (question) => {
```

Comment:
@jayesh9747 optimize this function 
sample
const getDateTemplate = (question) => {
  const template = {};
  const dateFormat = question?.dateFormat?.replace("-", "/") || "";
  const { date: questionDate } = questionTemplate;
  const { date: questionStaticDate } = questionStatic;

  console.log("getDate");

  for (const key of Object.keys(questionDate)) {
    const keyInLowerCase = key.toLowerCase();

    if (questionStaticDate.includes(key)) {
      template[key] = questionDate[key];
    } else {
      switch (keyInLowerCase) {
        case "interactiontypes":
          template[key] = [question[questionDate[key]]];
          break;

        case "body":
          template[key] = getQuestionBodyParagraph(question[questionDate[key]]);
          break;

        case "editorstate":
          template[key] = {
            question: getQuestionBodyParagraph(question[questionDate.body]),
          };
          break;

        case "interactions":
          template[key] = {
            validation: {
              required: question?.validation?.required ? "Yes" : "No",
            },
            response1: {
              validation: {
                pattern: dateFormat || question?.dateFormat,
              },
              autoCapture: question?.autoCapture,
            },
          };
          break;

        case "evidence":
          template[key] = question?.file
            ? { ...question.file, mimeType: question.file.type }
            : { mimeType: [] };
          template.showEvidence = question?.file ? "Yes" : undefined;
          break;

        case "instructions":
          template[key] = { default: question?.tip };
          break;

        case "showremarks":
          template[key] = question[questionDate[key]] ? "Yes" : "No";
          break;

        case "name":
          template[key] =
            question[questionDate[key]]?.length > 0
              ? question[questionDate[key]][0]
              : "Question";
          break;

        default:
          template[key] = question[questionDate[key]] || "";
          break;
      }
    }
  }

  return template;
};